OrbitQt/main.cpp

```diff
@@ -117,44 +133,75 @@ int main(int argc, char* argv[]) {
   app.setStyleSheet(
       "QToolTip { color: #ffffff; background-color: #2a82da; border: 1px solid "
       "white; }");
+}
 
-  if (options.asio_server_address.empty()) {
-    OrbitStartupWindow sw;
-    OrbitSsh::Credentials ssh_credentials;
-    int dialog_result = sw.Run(&ssh_credentials);
-    if (dialog_result == 0) return 0;
-
-#ifdef __linux__
-    options.asio_server_address =
-        absl::StrFormat("%s:%d", ssh_credentials.host, asio_port);
-    options.grpc_server_address =
-        absl::StrFormat("%s:%d", ssh_credentials.host, grpc_port);
-#else
-    // TODO(antonrohr) remove this ifdef as soon as the collector works on
-    // windows
-    if (ssh_credentials.host != "127.0.0.1") {
-      options.asio_server_address =
-          absl::StrFormat("%s:%d", ssh_credentials.host, asio_port);
-      options.grpc_server_address =
-          absl::StrFormat("%s:%d", ssh_credentials.host, grpc_port);
-    }
-#endif
-  }
-
-  OrbitMainWindow w(&app, std::move(options));
-
-  if (!w.IsHeadless()) {
-    w.showMaximized();
+static OrbitQt::DeploymentConfiguration figureOutDeploymentConfiguration() {
+  auto env = QProcessEnvironment::systemEnvironment();
```

Comment:
Just because im interested, why environment variables and not command line arguments?

Reply:
Eventually I want to have both, but because of priorities there is only one mechanism.

I went with env variables because it's more convenient. You set them once in your IDE or shell and every call of Orbit will find the values. Furthermore env variables are a bit more suitable for keeping passwords. Having passwords as a command line option will be visible to every user on the system (at least on Linux).